estimagic/dashboard/monitoring_app.py

```diff
@@ -197,110 +106,182 @@ def _map_groups_to_params(params):
 
-    Args:
-        attr: Required by bokeh.
-        old: Old state of the Button.
-        new: New state of the Button.
+    They will be periodically updated from the database.
+    There is a ColumnDataSource for all parameters and one for the criterion value.
+    The "x" column is called "iteration".
 
-        doc (bokeh.Document)
-        database (sqlalchemy.MetaData)
-        session_data (dict): This app's entry of infos to be passed between and within
-            apps. The keys are:
-            - last_retrieved (int): last iteration currently in the ColumnDataSource
-            - database_path
-        rollover (int): Maximal number of points to show in the plot.
-        tables (list): List of table names to load and convert to ColumnDataSources.
+    Args:
         start_params (pd.DataFrame): See :ref:`params`
 
+    Returns:
+        criterion_history (bokeh.ColumnDataSource)
+        params_history (bokeh.ColumnDataSource)
+
     """
-    callback_dict = session_data["callbacks"]
-    if new is True:
-        plot_new_data = partial(
-            _update_monitoring_tab,
-            doc=doc,
+    crit_data = {"iteration": [], "criterion": []}
+    criterion_history = ColumnDataSource(crit_data, name="criterion_history_cds")
+
+    param_names = start_params["name"].tolist()
+    params_data = {"iteration": []}
+    for name in param_names:
+        params_data[name] = []
+    params_history = ColumnDataSource(params_data, name="params_history_cds")
+
+    return criterion_history, params_history
+
+
+def _set_last_retrieved(session_data, database, rollover, jump):
+    """Set the last retrieved value.
+
+    As the session_data dictionary is used to communicate between apps, this is done
+    inplace.
+
+    Args:
+        session_data (dict): Infos to be passed between and within apps.
+            Keys of this app's entry are:
+            - last_retrieved (int): last iteration currently in the ColumnDataSource.
+            - database_path (str or pathlib.Path)
+            - callbacks (dict): dictionary to be populated with callbacks.
+        database (sqlalchemy.MetaData): Bound metadata object.
+        rollover (int): Upper limit to how many iterations are displayed.
+        jump (bool): If True jump to the last rollover iterations
+
+    """
+    if jump:
+        last_entry = read_last_rows(
             database=database,
-            session_data=session_data,
-            rollover=rollover,
-            tables=tables,
-            start_params=start_params,
-        )
-        callback_dict["plot_periodic_data"] = doc.add_periodic_callback(
-            plot_new_data, period_milliseconds=200
+            table_name="optimization_iterations",
+            n_rows=1,
+            return_type="list_of_dicts",
         )
-        # change the button color
-        button.button_type = "success"
-        button.label = "Reset Plot"
+        session_data["last_retrieved"] = max(0, last_entry[0]["rowid"] - rollover)
```

Comment:
This does not exactly do what you describe for the `--jump` option. It does not directly generate a plot with the last rollover observations but jumps to the first of those observations and then updates with the standard update-chunk.

Reply:
Addressed in 9f3c5cbb9b9c3fe40548ed30da551737dfee5523.